File: web/src/locales/centralized-mgmt/en-US.ts
/**
 * Centralized Management i18n - English (US)
 */
export default {
  // Tab label
  tabLabel: "Centralized Management",

  // Endpoint display
  supportedChannels: "Supported Channels",
  channelHint:
    "Requests are forwarded to groups/aggregates for processing (chat, audio, image, video, etc.)",
  copyBaseUrl: "Copy Base URL",
  baseUrlCopied: "Base URL copied",
  endpointCopied: "Endpoint URL copied",

  // Model pool
  modelPool: "Model Pool",
  modelName: "Model Name",
  sourceGroups: "Source Groups",
  groupType: "Group Type",
  channelType: "Channel Type",
  healthScore: "Health Score",
  effectiveWeight: "Effective Weight",
  groupCount: "Groups",
  standardGroup: "Standard",
  subGroup: "Sub",
  aggregateGroup: "Aggregate",
  aggregateGroupShort: "A",
  subGroupShort: "S",
  noModels: "No models available",
  noEnabledGroups: "No enabled groups",
  searchModelPlaceholder: "Search model or group name...",
  totalModels: "{total} models",
  filterAll: "All",
  editPriority: "Edit Priority",

  // Priority
  priority: "Priority",
  priorityHint: "1-999=priority (lower=higher priority), 1000=internal reserved value",
  prioritySortHint:
    "Within the same priority, higher health score = higher selection probability. E.g., two priority=10 groups are weighted by health",
  priorityColumnHint:
    "Lower value = Higher priority (1=highest, 999=lowest, 1000=internal reserved)",
  priorityExplanationHint:
    "💡 Numbers on group tags (e.g., :20, :100) are priorities. Lower value = higher priority. Same priority uses health-weighted random selection",

  // Hub settings
  hubSettings: "Hub Settings",
  maxRetries: "Max Retries",
  maxRetriesHint: "Max retries within the same priority level",
  retryDelay: "Retry Delay",
  healthThreshold: "Health Threshold",
  healthThresholdHint: "Groups below this threshold will be skipped",
  enablePriority: "Enable Priority Routing",
  onlyAggregateGroups: "Only Aggregate Groups",
  onlyAggregateGroupsHint:
    "When enabled, Hub only routes to aggregate groups, ignoring standard groups",

  // Access keys
  accessKeys: "Access Keys",
  accessKeyName: "Name",
  maskedKey: "Key",
  keyCopied: "Key copied",
  keyNameCopied: "Name copied",
  allowedModels: "Allowed Models",
  allModels: "All Models",
  specificModels: "{count} models",
  createAccessKey: "Create Access Key",
  editAccessKey: "Edit Access Key",
  deleteAccessKey: "Delete Access Key",
  confirmDeleteAccessKey: 'Are you sure you want to delete access key "{name}"?',
  accessKeyCreated: "Access key created successfully",
  accessKeyUpdated: "Access key updated successfully",
  accessKeyDeleted: "Access key deleted successfully",
  accessKeyToggled: "Access key status updated",
  noAccessKeys: "No access keys",
  keyCreatedCopyHint: "Please copy and save the key, it will only be shown once",
  keyOnlyShownOnce:
    "The key is only shown once at creation. It cannot be viewed again after closing. Please make sure you have copied it.",

  // Access key form
  keyName: "Key Name",
  keyNamePlaceholder: "Enter key name",
  keyValue: "Key Value",
  keyValuePlaceholder: "Leave empty to auto-generate, or enter custom key",
  keyValueHint: "Leave empty to auto-generate a key with hk- prefix",
  allowedModelsMode: "Model Permissions",
  allowedModelsModeAll: "Allow access to all models",
  allowedModelsModeSpecific: "Allow access to specific models only",
  selectAllowedModels: "Select Allowed Models",
  searchModelsPlaceholder: "Search models...",
  selectedModelsCount: "{count} models selected",

  // Panel
  centralizedManagement: "Centralized Management",
  onlyAggregateGroupsActive:
    "🔒 Only Aggregate Groups Mode Enabled (Can be changed in Hub Settings)",
  refreshModelPool: "Refresh Model Pool",
  refreshing: "Refreshing...",
  totalAccessKeys: "{total} keys",

  // Usage statistics
  usageCount: "Usage Count",
  lastUsedAt: "Last Used",
  neverUsed: "Never used",
  justNow: "Just now",
  minutesAgo: "{n} minutes ago",
  hoursAgo: "{n} hours ago",
  daysAgo: "{n} days ago",
  monthsAgo: "{n} months ago",
  yearsAgo: "{n} years ago",

  // Batch operations
  batchOperations: "Batch Operations",
  batchDelete: "Batch Delete",
  batchEnable: "Batch Enable",
  batchDisable: "Batch Disable",
  selectedKeys: "{count} keys selected",
  confirmBatchDelete: "Are you sure you want to delete {count} selected access keys?",
  batchDeleteSuccess: "Successfully deleted {count} access keys",
  batchEnableSuccess: "Successfully enabled {count} access keys",
  batchDisableSuccess: "Successfully disabled {count} access keys",
  selectAtLeastOne: "Please select at least one access key",

  // Custom models
  customModels: "Custom Models",
  customModelNames: "Custom Model Names",
  customModelNamesHint: "Add custom model names for aggregate groups, one per line",
  addCustomModel: "Add Model",
  editCustomModels: "Edit Custom Models",
  noCustomModels: "No custom models",
  customModelsUpdated: "Custom models updated",
  aggregateGroupName: "Aggregate Group",
  modelCount: "{count} models",
  customModelBadge: "Custom",
  customModelTooltip: "This is a user-defined custom model name",

  // Routing logic
  routingLogic: "Routing Logic (Sequential)",
  routingStep1:
    "① Path → Format Detection (Chat/Claude/Gemini/Image/Audio). Unknown formats fallback to OpenAI",
  routingStep2: "② Extract model name from request",
  routingStep3: "③ Access Control: Validate key permissions",
  routingStep4: "④ Model Availability: Check if model exists in any enabled group",
  routingStep5:
    "⑤ Group Selection Filters: Health threshold + Enabled status + Channel compatibility + CC support (Claude) + Aggregate preconditions (request size limits, etc.)",
  routingStep6: "⑥ Channel Priority: Native channel > Compatible channel",
  routingStep7: "⑦ Group Selection: Min priority value (lower=higher) → Health-weighted random",
  routingStep8: "⑧ Path Rewrite & Forward: /hub/v1/* → /proxy/group-name/v1/*",
};

File: internal/centralizedmgmt/i18n/en-US.go
// Package i18n provides internationalization support for centralized management
package i18n

// MessagesEnUS contains English translations for centralized management
var MessagesEnUS = map[string]string{
	// Hub access key related
	"hub.access_key.created":      "Hub access key created successfully",
	"hub.access_key.updated":      "Hub access key updated successfully",
	"hub.access_key.deleted":      "Hub access key deleted successfully",
	"hub.access_key.not_found":    "Hub access key not found",
	"hub.access_key.invalid":      "Invalid hub access key",
	"hub.access_key.disabled":     "Hub access key is disabled",
	"hub.access_key.name_exists":  "Hub access key name already exists",
	"hub.access_key.key_required": "Access key is required",

	// Hub model pool related
	"hub.model_pool.updated":           "Model pool updated successfully",
	"hub.model_pool.priority_updated":  "Model group priority updated successfully",
	"hub.model_pool.invalid_priority":  "Priority must be between 1 and 999 (1000 is reserved for internal use)",
	"hub.model_pool.model_not_found":   "Model not found in pool",
	"hub.model_pool.no_healthy_groups": "No healthy groups available for model",

	// Hub settings related
	"hub.settings.updated":              "Hub settings updated successfully",
	"hub.settings.invalid_threshold":    "Health threshold must be between 0 and 1",
	"hub.settings.invalid_retry_config": "Invalid retry configuration",

	// Hub routing related
	"hub.routing.model_required":         "Model is required in request",
	"hub.routing.model_not_allowed":      "Model not allowed by access key",
	"hub.routing.model_not_available":    "Model not available in any group",
	"hub.routing.group_selection_failed": "Failed to select group for model",
	"hub.routing.no_healthy_group":       "No healthy groups available for model",

	// Hub routing logic description
	"hub.routing.logic.title":       "Hub Routing Logic",
	"hub.routing.logic.description": "Request routing follows these steps in order",
	"hub.routing.logic.step1":       "① Path Format Detection: Identify API format (Chat/Claude/Gemini/Image/Audio). Unknown formats fallback to OpenAI.",
	"hub.routing.logic.step2":       "② Model Extraction: Extract model name from request (format-aware)",
	"hub.routing.logic.step3":       "③ Access Control: Validate access key permissions for the model",
	"hub.routing.logic.step4":       "④ Model Availability: Check if model exists in any enabled group",
	"hub.routing.logic.step5":       "⑤ Group Selection Filters: Health threshold + Enabled status + Channel compatibility + Claude Code support + Aggregate group preconditions (request size limits, etc.)",
	"hub.routing.logic.step6":       "⑥ Channel Priority: Native channels > Compatible channels",
	"hub.routing.logic.step7":       "⑦ Group Selection: Minimum priority value (lower=higher) → Health-weighted random selection",
	"hub.routing.logic.step8":       "⑧ Path Rewrite & Forward: /hub/v1/* → /proxy/{group_name}/v1/*",
	"hub.routing.logic.note":        "Note: Model matching happens first to determine the available group range, then path format is used for channel compatibility filtering.",

	// Channel types
	"channel.type.openai":    "OpenAI",
	"channel.type.anthropic": "Anthropic",
	"channel.type.gemini":    "Gemini",
	"channel.type.codex":     "Codex",
	"channel.type.azure":     "Azure",
	"channel.type.custom":    "Custom",

	// Relay formats
	"relay_format.openai_chat":                "OpenAI Chat Completions",
	"relay_format.openai_completion":          "OpenAI Completions",
	"relay_format.claude":                     "Claude Messages",
	"relay_format.codex":                      "Codex Responses",
	"relay_format.openai_image":               "OpenAI Image Generation",
	"relay_format.openai_image_edit":          "OpenAI Image Editing",
	"relay_format.openai_audio_transcription": "OpenAI Audio Transcription",
	"relay_format.openai_audio_translation":   "OpenAI Audio Translation",
	"relay_format.openai_audio_speech":        "OpenAI Text-to-Speech",
	"relay_format.openai_embedding":           "OpenAI Embeddings",
	"relay_format.openai_moderation":          "OpenAI Moderation",
	"relay_format.gemini":                     "Gemini",
	"relay_format.unknown":                    "Unknown Format (defaults to OpenAI)",

	// Endpoint descriptions
	"endpoint.chat_completions":     "Chat Completions",
	"endpoint.completions":          "Text Completions",
	"endpoint.messages":             "Messages",
	"endpoint.responses":            "Responses",
	"endpoint.images_generations":   "Image Generation",
	"endpoint.images_edits":         "Image Editing",
	"endpoint.images_variations":    "Image Variations",
	"endpoint.audio_transcriptions": "Audio Transcription",
	"endpoint.audio_translations":   "Audio Translation",
	"endpoint.audio_speech":         "Text-to-Speech",
	"endpoint.embeddings":           "Embeddings",
	"endpoint.moderations":          "Content Moderation",
}
